File: app/src/main/java/ru/zakoulov/gallery/imageController/tasks/LoadImages.java
package ru.zakoulov.gallery.imageController.tasks;

import android.os.AsyncTask;
import android.util.Log;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.List;
import java.util.Locale;

import ru.zakoulov.gallery.imageController.Image;
import ru.zakoulov.gallery.imageController.ImageController;

/**
 * Created by Илья on 27.04.2018.
 */
public class LoadImages extends AsyncTask<String, Void, List> {
    int countImages;
    Date date;
    String rootPath;

    @Override
    protected List doInBackground(String[] params) {
        List<Image> images = new ArrayList<>();
        Calendar calendar = new GregorianCalendar();
        calendar.setTime(date);
        SimpleDateFormat sdf = new SimpleDateFormat("yyMMdd", Locale.getDefault());
        String url;
        try {
            while (images.size() < countImages) {
                Image image = new Image();
                url = rootPath + "ap" + sdf.format(calendar.getTime()) + ".html";
                Log.d("loadImages", url);
                Log.d("loadImages", images.size() + "");
                Document doc = Jsoup.connect(url).get();
                Elements elements = doc.body().select("center").first().select("p").get(1).select("a");
                if (elements.size() == 0) {
                    calendar.add(Calendar.DAY_OF_YEAR, -1);
                    continue;
                }

                image.setDate(calendar.getTime());
                image.setPath(rootPath + elements.first().select("img").first().attr("src"));
                image.setFullPath(rootPath + elements.first().attr("href"));
                image.setName(doc.body().select("center").get(1).select("b").first().text());
                images.add(image);
                calendar.add(Calendar.DAY_OF_YEAR, -1);
            }
        } catch (IOException e) {
            Log.d("error", e.getMessage());
            ImageController.responseImagesDownload(null);
            return null;
        }
        ImageController.responseImagesDownload(images);
        return null;
    }

    public void setCountImages(int countImages) {
        this.countImages = countImages;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public void setRootPath(String rootPath) {
        this.rootPath = rootPath;
    }
}

File: app/src/main/java/ru/zakoulov/gallery/imageController/tasks/LoadImages.kt
package ru.zakoulov.gallery.imageController.tasks

import android.os.AsyncTask
import android.util.Log

import org.jsoup.Jsoup

import java.io.IOException
import java.text.SimpleDateFormat
import java.util.ArrayList
import java.util.Calendar
import java.util.Date
import java.util.GregorianCalendar
import java.util.Locale

import ru.zakoulov.gallery.imageController.Image

/**
 * Created by Илья on 27.04.2018.
 */
class LoadImages(val countImages: Int = 0,
                 val date: Date,
                 val rootPath: String,
                 val taskResponse: TaskResponseImages) : AsyncTask<String, Void, Void?>() {

    override fun doInBackground(params: Array<String>): Void? {
        val images = ArrayList<Image>()
        val calendar = GregorianCalendar()
        calendar.time = date
        val sdf = SimpleDateFormat("yyMMdd", Locale.getDefault())
        var url: String
        try {
            while (images.size < countImages) {
                url = rootPath + "ap" + sdf.format(calendar.time) + ".html"
                Log.d("loadImages", url)
                Log.d("loadImages", images.size.toString() + "")
                val doc = Jsoup.connect(url).get()
                val elements = doc.body().select("center").first().select("p")[1].select("a")
                Log.d("loadImages", elements.toString() + "")
                if (elements.size == 0 || !elements.first().parent().`is`("p")) {
                    calendar.add(Calendar.DAY_OF_YEAR, -1)
                    continue
                }
                val image = Image(
                    date=calendar.time,
                    path=rootPath+elements.first().select("img").first().attr("src"),
                    fullPath=rootPath+elements.first().attr("href"),
                    name=doc.body().select("center")[1].select("b").first().text()
                )
                images.add(image)
                calendar.add(Calendar.DAY_OF_YEAR, -1)
            }
        } catch (e: IOException) {
            Log.d("error", e.message)
            taskResponse.responseImagesError()
            return null
        }
        taskResponse.responseImagesDownload(images)
        return null
    }
}
